### PR TITLE
(#638) Enable multiple Sensu Enterprise Dashboard API endpoints

### LINKED ISSUE #584: Unable to add ssl and insecure Sensu attributes to API section of dashboard.json
## Description of problem

- What did you do?
Attempted to use enterprise dashboard API definition with SSL and insecure Puppet attributes.
- What happened?
This error during application of the configuration on my Sensu server:
```
Error: /Stage[main]/Ao_sensu::Server/Sensu::Enterprise::Dashboard::Api[MyDatacenter]/Sensu_enterprise_dashboard_api_config[MyDatacenter]: Could not evaluate: undefined method `retrieve' for #<Puppet::Type::Sensu_en
terprise_dashboard_api_config::Ssl:0x0000000441e958>
```
Same error happens for insecure.
- What did you expect to happen?
Puppet adding the SSL and insecure Sensu attributes to Sensu's dashboard.json config file:
```
  "sensu": [
    {
      "name": "MyDatacenter",
      "host": "example.com",
      "port": 4568,
      "ssl": true,
      "insecure": true,
      "user": "secret",
      "pass": "secret",
      "timeout": 5
    }
  ]
```
- How can someone reproduce the problem?
Call the sensu enterprise dashboard api definition like this:

```
  sensu::enterprise::dashboard::api { 'MyDatacenter':
    host     => 'example.com',
    port     => 4568,
    ssl      => true,
    insecure => true,
    user     => secret,
    pass     => secret,
  }
```
Then observe the error during an agent run.
## Command used and debugging output

- What does your puppet config look like (including any hiera config)
See above for reproducing the issue. I do use hiera, but I've simplied it during troubleshooting to just use Puppet for now. Let me know if more code details are needed.
- Is this a masterless or master based puppet setup?
Master based.

## Platform and version information

- Your OS: CentOS release 6.8 (Puppet master) and CentOS Linux release 7.2.1511 (Sensu server)
- Your Ruby version:ruby 1.8.7 (2013-06-27 patchlevel 374) [x86_64-linux]
- Your version of Puppet: PE 4.7.0
- Your version of Sensu: 26.3 (Sensu Enterprise 1.14.8)
- Your version of this module: 2.1.0

## Anything else to add that you think will be helpful?
Entirely possible I am just not using the module correctly here, but I couldn't find another way to get this config generated and the readme doesn't include examples down this far. I based my code off reading the types screen on Puppet Forge and looking at the manifests.

The code does work as expected with both the ssl and insecure attributes removed from my Puppet code.

### LINKED ISSUE #638: sensu_enterprise_dashboard_api type should use `host` as namevar, not `name`
Because `name` is used as namevar, it is not possible to have multiple API endpoints represented as a single datacenter in the enterprise dashboard.

### PR DESCRIPTION
**Pull request by jeffmccune**
Without this patch it is not possible to manage multiple
sensu::enterprise::dashboard::api resources in the same datacenter because
the resources conflict over the namevar.  This patchset addresses the
problem by changing the namevar to the `host` and establishing the
`datacenter` attribute.

An example of how to exercise this behavior is located in
tests/sensu-enterprise-dashboard-multi-api.pp

resolves #638
closes #651
closes #584

### CODE REVIEW

File: tests/sensu-enterprise-dashboard-multi-api.pp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Usage:
+# Bring up the sensu-server Vagrant vm.
+# After provisioning completes, run:
+# puppet apply -v /vagrant/tests/sensu-dashboard-api.pp
+node 'sensu-server' {
+  # This configuration is expected to produce /etc/sensu/dashboard.json with the values of:
+  # name: example-dc
+  # host: sensu.example.com
+
+  if ! ($facts['se_user'] or $facts['se_pass']) {
```
Comment (ghoneycutt) [inline tests/sensu-enterprise-dashboard-multi-api.pp]: are these facts already there or do they have to be exported somehow?

**Comment by ghoneycutt** [comment]
Awesome work!

**Comment by ghoneycutt** [comment]
Looks like style issues with the new test manifest

**Comment by jeffmccune** [comment]
Dooh, fixing now.

**Comment by jeffmccune** [comment]
Fixed, passing locally.  Should be good.

**Comment by ghoneycutt** [comment]
Released in v2.14.0

**Pull request merged**